### PR TITLE
Simplify options for process-historical-blocks

### PR DESCRIPTION
**Pull request by geoff-vball**
## Why this should be merged
Previously, we were using `process-missed-blocks` to determine whether to process historical AND missed blocks. This PR splits those out into two different config options.

## How this works
`process-historical-blocks` defaults to `false` will only work if `start-block-height` is set, and will start processing blocks from the greater of `start-block-height` and the last processed block height (if it exists in the db).

`process-missed-blocks` defaults to `true`, and will start processing blocks from the last processed block height (if it exists in the db). If there is no latest block in the db, or `process-missed-blocks` is `false`, we will start processing from the chain head.

## How is this documented
README

### CODE REVIEW

**Comment by martineckardt** [comment]
Idea: Why don't we remove the option `ProcessHistoricalBlocks` from the global config and process historical blocks if the `StartBlockHeight` optional value is set in the sourceBlockchain config?

If I understand correctly, if the `ProcessHistoricalBlocks` option is set to `true` then every sourceBlockchain needs a `StartBlockHeight`. This means the relayer operator cannot decide this on a blockchain-by-blockchain basis. 

With omitting `ProcessHistoricalBlocks` from the global config we additionally eliminate the scenario where a relayer operator set's the `StartBlockHeight`, but forgets to also set `ProcessHistoricalBlocks` option is set to `true`. 

To make it more intuitive, I would rename the `"start-block-height"` option to something like `"process-historical-blocks-from-height"`.

**Comment by geoff-vball** [comment]
@martineckardt I did actually just have that thought as I pushed the last update here. I'll make that change now.

File: config/config.go [hunk 1]
```diff
@@ -63,7 +61,7 @@ type SourceBlockchain struct {
 	WSEndpoint            string                           `mapstructure:"ws-endpoint" json:"ws-endpoint"`
 	MessageContracts      map[string]MessageProtocolConfig `mapstructure:"message-contracts" json:"message-contracts"`
 	SupportedDestinations []string                         `mapstructure:"supported-destinations" json:"supported-destinations"`
-	StartBlockHeight      uint64                           `mapstructure:"start-block-height" json:"start-block-height"`
+	StartBlockHeight      uint64                           `mapstructure:"process-historical-blocks-from-height" json:"process-historical-blocks-from-height"`
```
Comment (cam-schultz) [inline config/config.go:64]: Let's update this variable name

File: relayer/relayer.go [hunk 2]
```diff
@@ -174,7 +169,7 @@ func NewRelayer(
 		// ProcessFromHeight to overload the message queue and cause a deadlock.
 		go sub.ProcessFromHeight(big.NewInt(0).SetUint64(height), r.catchUpResultChan)
 	} else {
-		err = r.setProcessedBlockHeightToLatest()
+		_, err = r.setProcessedBlockHeightToLatest()
```
Comment (cam-schultz) [inline relayer/relayer.go:172]: Can we add a log here that we're initializing the relayer from the latest height?

File: relayer/relayer.go [hunk 3]
```diff
@@ -306,15 +303,15 @@ func (r *Relayer) calculateStartingBlockHeight(startBlockHeight uint64) (uint64,
 	return startBlockHeight, nil
 }
 
-func (r *Relayer) setProcessedBlockHeightToLatest() error {
+func (r *Relayer) setProcessedBlockHeightToLatest() (uint64, error) {
```
Comment (cam-schultz) [inline relayer/relayer.go:306]: Can you add a doccomment to this function explaining the return values?

File: relayer/relayer.go [hunk 4]
```diff
@@ -174,7 +169,11 @@ func NewRelayer(
 		// ProcessFromHeight to overload the message queue and cause a deadlock.
 		go sub.ProcessFromHeight(big.NewInt(0).SetUint64(height), r.catchUpResultChan)
 	} else {
-		err = r.setProcessedBlockHeightToLatest()
+		r.logger.Info(
+			"processed-missed-blocks set to false, starting process from chain head",
```
Comment (cam-schultz) [inline relayer/relayer.go:173]:
```suggestion
			"processed-missed-blocks set to false, starting processing from chain head",
```

**Review by cam-schultz — APPROVED** [review]
LGTM

**Review by michaelkaplan13 — APPROVED** [review]
Makes sense to me

**Pull request merged**